### PR TITLE
add alert for oom for services

### PR DESCRIPTION
**Pull request by k4leung4**
applied here, [link](https://console.cloud.google.com/monitoring/dashboards/builder/f7a35b30-7561-4fbf-bbe3-500d9c9dab52;duration=P1D?project=kleung-chainguard&supportedpurview=project&f.rlabel.service_name=enforce-registry)

### CODE REVIEW

File: modules/dashboard/sections/resources/main.tf [hunk 1]
```diff
@@ -55,6 +60,39 @@ module "received_bytes" {
...
+  }
+
+  display_name = "${var.service_name} OOM Alert"
+  combiner     = "OR"
+
+  conditions {
+    display_name = "${var.service_name} OOM Alert"
+
+    condition_matched_log {
+      filter = "${join(" AND ", var.filter)} AND logName:\"run.googleapis.com%2Fvarlog%2Fsystem\" AND resource.labels.service_name=\"enforce-staging-ing-sbom\" AND severity=ERROR and textPayload:\"Consider increasing the memory limit\""
```
Comment (mattmoor) [inline modules/dashboard/sections/resources/main.tf:86]: hardcoded service_name

Comment (k4leung4) [reply modules/dashboard/sections/resources/main.tf]: fixed.

File: modules/dashboard/sections/resources/main.tf [hunk 2]
```diff
@@ -104,6 +142,14 @@ locals {
       height = local.unit,
       width  = local.unit,
       widget = module.received_bytes.widget,
+    },
+    {
+      yPos   = local.unit * 2,
```
Comment (mattmoor) [inline modules/dashboard/sections/resources/main.tf:147]: shift others down, and make this the top row with `module.width.size` ?

File: modules/dashboard/sections/resources/main.tf [hunk 3]
```diff
@@ -55,6 +60,39 @@ module "received_bytes" {
...
+
+  display_name = "${var.cloudrun_name} OOM Alert"
+  combiner     = "OR"
+
+  conditions {
+    display_name = "${var.cloudrun_name} OOM Alert"
+
+    condition_matched_log {
+      filter = "${join(" AND ", var.filter)} AND logName:\"run.googleapis.com%2Fvarlog%2Fsystem\" AND severity=ERROR AND textPayload:\"Consider increasing the memory limit\""
+
```
Comment (mattmoor) [inline modules/dashboard/sections/resources/main.tf:87]: empty line

File: modules/dashboard/sections/resources/main.tf [hunk 4]
```diff
@@ -104,6 +142,14 @@ locals {
...
       width  = local.unit,
       widget = module.received_bytes.widget,
+    },
+    {
+      yPos   = local.unit * 2,
+      xPos   = local.col[0],
+      height = local.unit,
+      width  = local.unit,
+      widget = module.oom_alert.widget,
+
```
Comment (mattmoor) [inline modules/dashboard/sections/resources/main.tf:152]: empty line

Comment (k4leung4) [reply modules/dashboard/sections/resources/main.tf]: fixed.

File: modules/dashboard/sections/resources/main.tf (hunk 2 above)
Comment (k4leung4) [reply modules/dashboard/sections/resources/main.tf]: fixed.

**Review by mattmoor — APPROVED** [review]

**Pull request merged**